proposals/ssss-distributed-actor-runtime.md

```diff
@@ -1348,44 +1392,49 @@ extension DistributedActorSystem {
 
-#### Performing the distributed target call
+#### Recipient: Executing the distributed target
 
-Once we have completed all the above steps, all building up to actually invoking the target of a remote call: it is finally time to do so, by calling the `executeDistributedTarget` method:
+Now that we have completed all the above steps, all building up to actually invoking the target of a remote call: it is finally time to do so, by calling the `executeDistributedTarget` method:
 
 ```swift
 // inside recipient actor system
-let envelope: Envelope = // receive & decode ...
+let envelope: IncomingEnvelope = // receive & decode ...
 let recipient: DistributedActor = // resolve ...
 
-// FIXME: make a separate type for DECODING side (!!!!!!!)
-let invocation = makeInvocation(from: envelope)
+let invocationDecoder = InvocationDecoder(system: self, bytes: envelope.bytes)
 
 try await executeDistributedTarget(
   on: recipient, // target instance for the call
   mangledName: envelope.targetName, // target func/var for the call
   invocation: invocation // will be used to perform decoding arguments,
-  handler: ReplyHandler(to: envelope) // handles replying to the caller (omitted in proposal)
+  handler: ClusterTargetInvocationResultHandler(system, envelope) // handles replying to the caller (omitted in proposal)
 )
 ```
 
 This call triggers all the decoding that we discussed earlier, and if any of the decoding, or distributed func/var resolution fails this call will throw. Otherwise, once all decoding has successfully been completed, the arguments are passed through the buffer to a distributed method accessor that actually performs the local method invocation. Once the method returns, its results are moved into the handler where the actor system takes over in order to send a reply to the remote caller - completing the remote call!
 
-For sake of completeness, the listing below shows the distributed method accessor thunk that is synthesized by the compiler. The thunk is passed the memory buffer into which argument values have just been stored during the `decodeArgument` calls on the `Invocation`. The thunk unpacks the well-typed buffer that contains the method call arguments into local variables, and invokes the actual target, like this:
+For sake of completeness, the listing below shows the **distributed method accessor thunk** that is synthesized by the compiler. The thunk is passed the memory buffer into which argument values have just been stored during the `decodeNextArgument` calls on the `Invocation`. The thunk unpacks the well-typed buffer that contains the method call arguments into local variables, and invokes the actual target, like this:
 
 ```swift
 distributed actor DA {
```

Comment:
the `myComputeThunk` is outdated as well in this example.

Reply:
Thanks for spotting it, I updated it, now it is:


extension DA {
  // Distributed accessor thunk" for 'myCompute(_:_:_:) -> String'
  //
  // PSEUDO CODE FOR ILLUSTRATION PURPOSES; NOT IMPLEMENTABLE IN PLAIN SWIFT; 
  // Implemented in directly in IR for expressability reasons, and not user-accessible.
  nonisolated func $distributedFuncAccessor_myCompute(
    decoder: UnsafeMutableRawPointer, 
    argumentTypes: UnsafeRawPointer,
    resultBuffer: UnsafeRawPointer,
    genericSubstitutions: UnsafeRawPointer,
    witnessTables: UnsafeRawPointer,
    numWitnessTables: Int,
    actorSelf: UnsafeRawPointer) async {
    
    // - get generic signature of 'myCompute'
    // - create storage 'args' for all the parameters; it will be used directly
    // - for every argument, get the argumentType
    //   - invoke 'decoder.decodeArgument<Argument>()'
    //   - store in 'args'
    // - deal with the generic substitutions, witness tables and prepare the call
    // invoke 'myCompute' with 'args', and the prepared 'result' and 'error' buffers
  }
}